src/pages/MovieDetailPage.js

```diff
@@ -1,142 +1,155 @@
-            return;
-        }
-        async function fetchReviews() {
-            const url = `${API_URL}/movie/${id}/reviews?api_key=${API_KEY}&language=en-US&page=1`;
-            const response = await fetch(url);
-            const data = await response.json();
-            setReviews(data.results)
-        };
-        fetchReviews();
-      }, [id]);
+  useEffect(() => {
+    if (!id) {
+      return;
+    }
+    async function fetchRecommendMovie() {
+      const url = `${API_URL}/movie/${id}/recommendations?api_key=${API_KEY}&language=en-US&page=1`;
+      const response = await fetch(url);
+      const data = await response.json();
+      setRecommendMovies(data.results);
+    }
+    fetchRecommendMovie();
+  }, [id]);
 
-    const handleReview=()=>{
-      setShowModal(true)
+  useEffect(() => {
+    if (!id) {
+      return;
+    }
+    async function fetchSimilarMovie() {
+      const url = `${API_URL}/movie/${id}/similar?api_key=${API_KEY}&language=en-US&page=1`;
+      const response = await fetch(url);
+      const data = await response.json();
+      setSimilarMovies(data.results);
     }
+    fetchSimilarMovie();
+  }, [id]);
 
+  useEffect(() => {
+    if (!id) {
+      return;
+    }
+    async function fetchReviews() {
+      const url = `${API_URL}/movie/${id}/reviews?api_key=${API_KEY}&language=en-US&page=1`;
+      const response = await fetch(url);
+      const data = await response.json();
+      setReviews(data.results);
+    }
+    fetchReviews();
+  }, [id]);
 
-    return (
-        <div className="detailPage">
-            <NavBar/>
-            <Container className="mt-5">
-                {!loading && <MovieDetail 
-                path={movie.backdrop_path} 
-                title={movie.title} 
-                date={movie.release_date} 
-                vote={movie.vote_average}
-                overview={movie.overview}
-                genres={movie.genres}
-                clicked={handleReview}/>}
-            </Container>
-            <ModalReviews show={showModal} onHide={() => setShowModal(false)} reviews={reviews}/>
+  const handleReview = () => {
+    setShowModal(true);
+  };
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
 
-            {ids.length > 0 && (
-              <Container>
-              <ReactPlayer url={`https://www.youtube.com/watch?v=${ids[0].key}`} width='100%'/>
-              </Container>
-            )}
+  return (
+    <div className="detailPage">
+      <NavBar />
+      <Container className="mt-5">
+        {!loading && (
+          <MovieDetail
+            path={movie.backdrop_path}
+            title={movie.title}
+            date={movie.release_date}
+            vote={movie.vote_average}
+            overview={movie.overview}
+            genres={movie.genres}
+            clicked={handleReview}
+            handleClose={handleClose}
+            handleShow={handleShow}
+            show={show}
+            youtubeKey={key}
```

Comment:
Thanks Ms Annie so much. I can learn more about the importance of naming props in React ^^